Source/CSharpEssentials/GetterOnlyAutoProperty/UseGetterOnlyAutoPropertyAnalyzer.cs

```diff
@@ -1,35 +1,214 @@
+
+                    // Implicit assignment
+                    case SyntaxKind.OrAssignmentExpression:
+                    case SyntaxKind.AndAssignmentExpression:
+                    case SyntaxKind.ExclusiveOrAssignmentExpression:
+                    case SyntaxKind.AddAssignmentExpression:
+                    case SyntaxKind.SubtractAssignmentExpression:
+                    case SyntaxKind.MultiplyAssignmentExpression:
+                    case SyntaxKind.DivideAssignmentExpression:
+                    case SyntaxKind.ModuloAssignmentExpression:
+                    case SyntaxKind.LeftShiftAssignmentExpression:
+                    case SyntaxKind.RightShiftAssignmentExpression:
+
+                    // Prefix unary expression
+                    case SyntaxKind.PreIncrementExpression:
+                    case SyntaxKind.PreDecrementExpression:
+
+                    // Postfix unary expression
+                    case SyntaxKind.PostIncrementExpression:
+                    case SyntaxKind.PostDecrementExpression:
+                        return true;
+
+                    // Early loop termination
+                    case SyntaxKind.Block:
+                    case SyntaxKind.ExpressionStatement:
+                        return false;
+                }
             }
+
+            return false;
+        }
+
+        private static bool IsWithinConstructorOf(SyntaxNode node, INamedTypeSymbol type, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            // Are we in a constructor?
+            for (; node != null; node = node.Parent)
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.ConstructorDeclaration:
+                        // In a constructor. Is it the constructor for the type that contains the property?
+                        var constructorSymbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+                        return constructorSymbol != null && (object)constructorSymbol.ContainingType == type;
+
+                    // Early out cases. There are many others, but these are the common ones.
+                    case SyntaxKind.ClassDeclaration:
+                    case SyntaxKind.StructDeclaration:
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.PropertyDeclaration:
+                        return false;
+                }
+            }
+
+            return false;
+        }
+
+        private static Diagnostic CreateDiagnostic(ISymbol symbol, CancellationToken cancellationToken)
+        {
+            foreach (var reference in symbol.DeclaringSyntaxReferences)
+            {
+                // The span should be on the setter, but the symbol is for the whole property declaration.
+                var declarationNode = reference.GetSyntax(cancellationToken) as PropertyDeclarationSyntax;
```

Comment:
"as" cast without testing without testing for null. If we're confident that's a PropertyDeclarationSyntax, maybe just a direct cast?